content/rust-lld-on-stable-1.90.0/index.md

```diff
@@ -0,0 +1,63 @@
+
+There are different linkers, however, and the usual advice to improve linking times is to use one of these newer and faster linkers, like LLVM's [`lld`](https://lld.llvm.org/) or Rui Ueyama's [`mold`](https://github.com/rui314/mold).
+
+Some of Rust's wasm and aarch64 targets already use `lld` by default. When using rustup, rustc ships with a version of `lld` for this purpose. When CI builds LLVM to use in the compiler, it also builds the linker and packages it. It's referred to as `rust-lld` to avoid colliding with any `lld` already installed on the user's machine.
+
+Since improvements to linking times are substantial, it would be a good default to use in the most popular targets. This has been discussed for a long time, for example in issues [#39915](https://github.com/rust-lang/rust/issues/39915) and [#71515](https://github.com/rust-lang/rust/issues/71515).
+
+To expand our testing, we have enabled rustc to use `rust-lld` by default on nightly, [in May 2024](https://blog.rust-lang.org/2024/05/17/enabling-rust-lld-on-linux.html). No major issues have been reported since then.
+
+We believe we've done all the internal testing that we could, on CI, crater, on our benchmarking infrastructure and on nightly, and plan to enable `rust-lld` to be the linker used by default on `x86_64-unknown-linux-gnu` for stable builds in 1.90.0.
+
+#### Benefits
+
+While this also enables the compiler to use more linker features in the future, the most immediate benefit is much improved linking times.
+
+Here are more details from the ripgrep example mentioned above: for an incremental rebuild, linking is reduced 7x, resulting in a 40% reduction in end-to-end compilation times. For a from-scratch debug build, it is a 20% improvement.
+
+![Before/after comparison of a `ripgrep` incremental debug build](ripgrep-comparison.png)
+
+Most binaries should see some improvements here, but it's especially significant with e.g. bigger binaries, or for incremental rebuilds, or when involving debuginfo. These usually see bottlenecks in the linker.
+
+Here's [a link](https://perf.rust-lang.org/compare.html?start=b3e117044c7f707293edc040edb93e7ec5f7040a&end=baed03c51a68376c1789cc373581eea0daf89967&stat=instructions%3Au&tab=compile) to the complete results from our benchmarks.
+
+#### Possible drawbacks
+
+From our prior testing, we don't really expect issues to happen in practice. It is a drop-in replacement for the vast majority of cases, but `lld` is not _bug-for-bug_ compatible with GNU ld.
+
+In any case, using `rust-lld` can be disabled if any problem occurs: use the `-C linker-features=-lld` flag to revert to using the system's default linker.
+
+Some crates somehow relying on these differences could need additional link args, though we also expect this to be quite rare.
```

Comment:
Do we want to hear about these? I think per line 55 below ("let us know if you encounter problems") the answer is yes, but maybe worth duplicating the call for an issue here to make that clear?

Reply:
If it's something that we can fix ourselves yes (like `-nostart-stop-gc`). Otherwise, apart from being aware of issues (and/or redirect them to the llvm repo), we may not be able to help much. Still good to be aware so I'll add something ok.